In [1]:
#Step1: Gathering Data
import numpy as np
import pandas as pd

In [4]:
new=pd.read_csv('/content/Social_Network_Ads.csv')

In [5]:
new.head()

User ID  Gender  Age  EstimatedSalary  Purchased
0  15624510    Male   19            19000          0
1  15810944    Male   35            20000          0
2  15668575  Female   26            43000          0
3  15603246  Female   27            57000          0
4  15804002    Male   19            76000          0

In [29]:
data=new.drop(['User ID'],axis=1)

In [30]:
data.head()

Gender  Age  EstimatedSalary  Purchased
0    Male   19            19000          0
1    Male   35            20000          0
2  Female   26            43000          0
3  Female   27            57000          0
4    Male   19            76000          0

In [31]:
#step2: Data pre-processing where we will be using one hot encoding method
data['Gender']=data['Gender'].apply({'Male':1,'Female':0}.get)

In [32]:
data.head()

Gender  Age  EstimatedSalary  Purchased
0       1   19            19000          0
1       1   35            20000          0
2       0   26            43000          0
3       0   27            57000          0
4       1   19            76000          0

In [33]:
#step3: Dividing data into dependent and independent data
x=data[['Gender','Age','EstimatedSalary']]#independent variables

In [35]:
y=data[['Purchased']]#dependent variables

In [37]:
#Splitting the data into training and testing data
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.75)#75% of dataset is used for training and 25% is used for testing

CREATING ML MODEL USING KNN ALGORITHIM

In [39]:
from sklearn.neighbors import KNeighborsClassifier
knn=KNeighborsClassifier(n_neighbors=5)#n_neighbor is th value of k=5

In [40]:
#training the model using fit()
knn.fit(x_train,y_train)

/usr/local/lib/python3.10/dist-packages/sklearn/neighbors/_classification.py:215: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return self._fit(X, y)


KNeighborsClassifier()

In [41]:
y_pred=knn.predict(x_test)

In [42]:
print(y_pred)

[0 1 0 1 0 0 1 0 0 0 1 0 0 0 1 0 1 0 1 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 1 0 0
 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 1 0 0 0 0 1 1 1 0 1 0 1 1 0 0 0 0 0 0
 0 0 1 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 1 0 0 0 1 0 0 1 0 0 0 1 1
 0 0 0 0 0 1 1 0 0 0 1 1 0 0 0 0 0 1 1 0 1 0 1 0 1 1 0 0 0 0 0 1 1 0 1 0 0
 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 0 1 0 1 1 0 0 0 1 1 0 1 0 1 1 0 0 1 0 0 0 1
 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 1 0 1 0
 0 0 0 1 0 1 1 1 0 1 0 0 1 1 0 1 1 0 0 1 0 0 0 1 1 0 1 1 0 0 0 1 0 1 0 1 0
 0 0 0 0 0 0 1 1 1 1 0 0 0 0 1 0 1 1 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 1 0 0 1
 0 0 0 1]


0 indicates person did not purchases the item and 1 indicates person purchased the item

In [47]:
from sklearn.metrics import confusion_matrix,accuracy_score


In [48]:
cm=confusion_matrix(y_test,y_pred)

In [49]:
acc=accuracy_score(y_test,y_pred)

In [55]:
print(cm)

[[165  33]
 [ 37  65]]


# Here 165 is True positive and 65 is True Negative, Similarly 33 is false positive and 37 is false negative

In [54]:
print(acc)

0.7666666666666667


In [52]:
#predicting the purchase of a new customer
data_new={'Gender':1,'Age':19,'EstimatedSalary':78000}

In [43]:
#checking accuracy of ML model
accuracy=knn.score(x,y)

In [44]:
print('The accuracy of the ML model using knn classification is',accuracy)

The accuracy of the ML model using knn classification is 0.78


In [46]:
#predicting the purchase of a new customer
data_new={'Gender':1,'Age':19,'EstimatedSalary':78000}

In [56]:
#converting dictinary into data frame
index=[1]#serial number
my_data=pd.DataFrame(data_new,index)

In [57]:
my_data

Gender  Age  EstimatedSalary
1       1   19            78000

In [58]:
#predicting the amount of health insurance the new customer is going to get
y_pred=knn.predict(my_data)

In [59]:
print('the amount of health insurance the customer is going to get',y_pred)

the amount of health insurance the customer is going to get [0]


In [53]:
data.head()

Gender  Age  EstimatedSalary  Purchased
0       1   19            19000          0
1       1   35            20000          0
2       0   26            43000          0
3       0   27            57000          0
4       1   19            76000          0